src/NuGet.Services.EndToEnd/Support/Configuration/E2ESecretConfigurationReader.cs

```diff
@@ -17,6 +17,8 @@ public class E2ESecretConfigurationReader : IConfigurationRoot
         private ISecretReaderFactory _secretReaderFactory;
         private Lazy<ISecretInjector> _secretInjector;
 
+        public IEnumerable<IConfigurationProvider> Providers => throw new NotImplementedException();
```

Comment:
What is the purpose to introduce this?

Reply:
This is part of the Extensions 2.x IConfigurationRoot interface. We don't use it anywhere. This is the same "implementation" that was used in NuGet.Services.Configuration when we did the Extensions 2.x upgrade.